packages/util-stream/src/readableToAsyncIterable.ts

```diff
@@ -0,0 +1,42 @@
+import { Readable } from "stream";
+
+/**
+ * @internal
+ *
+ * Convert object stream piped in into an async iterable. This
+ * adaptor should be deprecated when Node stream iterator is stable.
+ * Caveat: this adaptor won't have backpressure to inwards stream
+ *
+ * Reference: https://nodejs.org/docs/latest-v11.x/api/stream.html#stream_readable_symbol_asynciterator
+ */
+export async function* readableToAsyncIterable<T>(readStream: Readable): AsyncIterable<T> {
+  let streamEnded = false;
+  let generationEnded = false;
+  const records = new Array<T>();
```

Comment:
```suggestion
  const records: T[] = [];
```

Reply:
can this function be implemented without this queue?

Reply:
I think the reason is the same as https://github.com/aws/aws-sdk-js-v3/pull/4787#discussion_r1221956497, you can't read the stream chunk-by-chunk _and_ yield, but I might be missing something